tests/tests_suite_test.go

```diff
@@ -104,9 +105,28 @@ var _ = SynchronizedBeforeSuite(testsuite.SynchronizedBeforeTestSetup, testsuite
 var _ = SynchronizedAfterSuite(testsuite.AfterTestSuiteCleanup, testsuite.SynchronizedAfterTestSuiteCleanup)
 
 var _ = AfterEach(func() {
-	testCleanup()
+	if !hasOncePerOrderedCleanupLabel() {
+		testCleanup()
+	}
+})
+
+var _ = AfterEach(OncePerOrdered, func() {
+	if hasOncePerOrderedCleanupLabel() {
+		testCleanup()
+	}
 })
 
+func hasOncePerOrderedCleanupLabel() bool {
+	oncePerOrderedCleanupText := decorators.OncePerOrderedCleanup[0]
+	oncePerOrderedCleanup, err := CurrentSpecReport().MatchesLabelFilter(oncePerOrderedCleanupText)
+	if err != nil {
+		fmt.Printf("Failed to parse labels %q\n", err)
```

Comment:
Use `GinkgoT().Errorf()`? IIUC this should fail the suite. But I think that is better than to silently swallow the error.